File: conectaBrenannd-api/conectaBrenannd-api/src/main/java/com/conecta/brenannd/service/IDependenteService.java
package com.conecta.brenannd.service;

import com.conecta.brenannd.entity.Dependente;

public interface IDependenteService {

	Dependente save(final Dependente entity);

	Dependente update(final long id, final Dependente entity);

	void delete(final long id);

}

File: conectaBrenannd-web/src/app/components/entity/dependente/services/idependente.service.ts
import { Observable } from 'rxjs';
import { Dependente } from '../models/dependente.models';

export interface IDependenteService {
  save(request: Dependente): Observable<Dependente>;

  update(id: number, request: Dependente): Observable<Dependente>;

  delete(id: number): Observable<void>;

  list(): Observable<Dependente[]>;

  findByID(id: number): Observable<Dependente>;

  listByUser(): Observable<Dependente[]>;
}
